### DISCUSSION TITLE
telegram-online-chart

### DISCUSSION THREAD

**Discussion by giscus** [AMD-NICK/blog.amd-nick.me#21] [Blog comments]
# telegram-online-chart

Есть у меня друг-наркоман. Принимает дозу соц. сетей внутриглазно каждую секунду, пока не спит или работает.

https://blog.amd-nick.me/telegram-online-chart

**Comment by vlab97** [comment]
Первый вариант скрипта работает. Онлайн/оффлайн пишет. А вот от августа 21 года бот не работает. Деятельность имитирует, а по факту про онлайн не оповещает.

**Reply by vlab97** [reply]
https://xgm.guru/p/programms/integration-lua-and-python
Ваш скрипт возможно будет адаптировать таким решением?

**Reply by AMD-NICK** [reply]
> https://xgm.guru/p/programms/integration-lua-and-python
> Ваш скрипт возможно будет адаптировать таким решением?

не знаю, не работал с той библиотекой. Проще уже на питоне переписать. А еще могу предложить попробовать это: https://github.com/Karlend/telespy

Мой друг написал альтернативу первому варианту, которая сразу правильный csv файл генерирует

**Reply by vlab97** [reply]
> > https://xgm.guru/p/programms/integration-lua-and-python
> 
> > Ваш скрипт возможно будет адаптировать таким решением?
> 
> 
> 
> не знаю, не работал с той библиотекой. Проще уже на питоне переписать. А еще могу предложить попробовать это: https://github.com/Karlend/telespy
> 
> 
> 
> Мой друг написал альтернативу первому варианту, которая сразу правильный csv файл генерирует

К сожалению альтернатива вашего друга не работает. CSV файл был создан, но правки в него не вносятся. И в самой консоли нет данных по онлайну/оффлайну.

**Reply by AMD-NICK** [reply]
> > > https://xgm.guru/p/programms/integration-lua-and-python
> > 
> > 
> > > Ваш скрипт возможно будет адаптировать таким решением?
> > 
> > 
> > не знаю, не работал с той библиотекой. Проще уже на питоне переписать. А еще могу предложить попробовать это: https://github.com/Karlend/telespy
> > Мой друг написал альтернативу первому варианту, которая сразу правильный csv файл генерирует
> 
> К сожалению альтернатива вашего друга не работает. CSV файл был создан, но правки в него не вносятся. И в самой консоли нет данных по онлайну/оффлайну.

У меня его скрипт работает уже дольше 2 недель без остановок. Возможно, ты неправильно прописал в конфиге данные для отслеживания

**Comment by NEK2018** [comment]
Добрый день!
Подскажите, что-то из представленных ботов в статье и в комментариях работает в 2025 г.?

**Reply by AMD-NICK** [reply]
Сейчас нет возможности проверить, сорри. Но если найдешь и отпишешь здесь, то буду очень рад

**Comment by Eugi6** [comment]
Кайф! Все работает, чатгпт на питоне написал форматирование логов в csv
Пока сделал статистику на тестовых данных, потом посмотрю свою) Единственное не совсем понимаю, почему в диаграмме динамических рядов накопление уходит в 0
![image](https://github.com/user-attachments/assets/edb872c4-b501-431b-aede-a4e3068e23f5)

**Reply by AMD-NICK** [reply]
Похоже, ты прошаренный чувачок, раз зашел так далеко. Насчет гугла не знаю, сам я перешел на Metabase. Возможно, и тебя заинтересует он больше

**Reply by AMD-NICK** [reply]
P.S. Круто, что ты написал в комменты. Люблю, когда сюда пишут :)

**Reply by Webalta17** [reply]
Привет. А этот трекер позволяет отслеживать время посещения, если пользователь скрыл последнее время посещения настройками приватности?

**Reply by shockpast** [reply]
> Привет. А этот трекер позволяет отслеживать время посещения, если пользователь скрыл последнее время посещения настройками приватности?

Думаю ты сам ответил на свой вопрос =)

**Reply by Webalta17** [reply]
А не знаешь как можно отследить последнее время посещения, если оно скрыто настройками?

**Reply by shockpast** [reply]
> А не знаешь как можно отследить последнее время посещения, если оно скрыто настройками?

Никак вероятнее всего, потому что зачастую скрывают для всех и видно только если человека добавляют в контакты.
